src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -266,6 +270,12 @@ public Match createMatch(User user, User otherUser) {
     // Yes, it would be absolutely possible to do without the Pair class...
     Pair<User, User> pair = new Pair<>(user, otherUser);
     match.setUserPair(pair);
+
+    Chat chat = new Chat();
+    chatRepository.saveAndFlush(chat);
+
+    // new Chat gets added
+    match.setChat(chat);
```

Comment:
I sort of don't like that the userService has a instance variable chatRepository for just this. Can you think of any other way to do it differently, e.g (if you want a matchService, we could also extract everything about matches), like making smaller methods, and then return the match , or the chat to the controller, and from there call the other Service that just saves and flushes it (and returns it), then call the third Service that does it for what that service is responsible of. 

I tried to do that when implementing the game logic: See for an example:
![image](https://user-images.githubusercontent.com/75019468/167311093-ca8b4f9b-5a2d-464d-9e7a-cf5bbe74ab9e.png), how I jump between gameService and userService. This might need some major refactoring though, we could create an issue for that (it also affects the other team members).
(Recommendation, Must)

Reply:
yes, i also did not like it. But because of all the tests already done for userService, i did not wanted to change it now. I will make an issue for it.